src/transformers/pipelines/document_question_answering.py

```diff
@@ -0,0 +1,459 @@
+                The maximum length of the total sentence (context + question) in tokens of each chunk passed to the
+                model. The context will be split in several chunks (using `doc_stride` as overlap) if needed.
+            max_question_len (`int`, *optional*, defaults to 64):
+                The maximum length of the question after tokenization. It will be truncated if needed.
+            handle_impossible_answer (`bool`, *optional*, defaults to `False`):
+                Whether or not we accept impossible as an answer.
+            lang (`str`, *optional*):
+                Language to use while running OCR. Defaults to english.
+            tesseract_config (`str`, *optional*):
+                Additional flags to pass to tesseract while running OCR.
+
+        Return:
+            A `dict` or a list of `dict`: Each result comes as a dictionary with the following keys:
+
+            - **score** (`float`) -- The probability associated to the answer.
+            - **start** (`int`) -- The start word index of the answer (in the OCR'd version of the input or provided
+              `word_boxes`).
+            - **end** (`int`) -- The end word index of the answer (in the OCR'd version of the input or provided
+              `word_boxes`).
+            - **answer** (`str`) -- The answer to the question.
+        """
+        if isinstance(question, str):
+            inputs = {"question": question, "image": image}
+            if word_boxes is not None:
+                inputs["word_boxes"] = word_boxes
+        else:
+            inputs = image
+        return super().__call__(inputs, **kwargs)
+
+    def preprocess(
+        self,
+        input,
+        padding="do_not_pad",
+        doc_stride=None,
+        max_question_len=64,
+        max_seq_len=None,
+        lang=None,
+        tesseract_config="",
+    ):
+        # NOTE: This code mirrors the code in question answering and will be implemented in a follow up PR
+        # to support documents with enough tokens that overflow the model's window
+        #        if max_seq_len is None:
+        #            # TODO: LayoutLM's stride is 512 by default. Is it ok to use that as the min
+        #            # instead of 384 (which the QA model uses)?
+        #            max_seq_len = min(self.tokenizer.model_max_length, 512)
+
+        if doc_stride is not None:
+            # TODO implement
+            # doc_stride = min(max_seq_len // 2, 128)
+            raise ValueError("Unsupported: striding inputs")
+
+        image = None
+        image_features = {}
+        if "image" in input:
+            image = load_image(input["image"])
+            if self.feature_extractor is not None:
+                image_features.update(self.feature_extractor(images=image, return_tensors=self.framework))
+            elif self.model_type == ModelType.Donut:
+                raise ValueError("If you are using a VisionEncoderDecoderModel, you must provide a feature extractor")
```

Comment:
If Donut is supported, then why is this PR called "... to support extractive QA"? Donut can do more than extractive, as it's a generative model. Only the LayoutLM family treats it in an extractive, BERT-like way.

Reply:
Good point, let me remove that from the title.

Reply:
This seems fixed now :)